src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -0,0 +1,136 @@
+        val targetGroupResponse = albClient.createTargetGroup(tgRequest)
+
+        println("Getting target group arn...")
+
+        val albTargetGroupRequest = albClient.describeTargetGroups(DescribeTargetGroupsRequest.builder().names("$user_name-target-group").build())
+        val albTargetGroupArn = albTargetGroupRequest.targetGroups()[0].targetGroupArn()
+
+        val pathPattern = PathPatternConditionConfig.builder().values("/$user_name/*").build()
+        val albRuleCondition :RuleCondition = RuleCondition.builder().field("path-pattern").pathPatternConfig(pathPattern).build()
+        val userTargetGroup = TargetGroupTuple.builder().targetGroupArn(albTargetGroupArn).build()
+        val forwardAction = ForwardActionConfig.builder().targetGroups(userTargetGroup).build()
+        val albRuleAction = Action.builder().type("forward").forwardConfig(forwardAction).build()
+
+        println("Creating listener rule...")
+
+        var rulesResponse = albClient.describeRules(DescribeRulesRequest.builder().listenerArn(albListenerResponse.listeners()[0].listenerArn()).build())
+        val createRule = albClient.createRule(CreateRuleRequest.builder().listenerArn(albListenerResponse.listeners()[0].listenerArn()).priority(getPriorityVal(rulesResponse)).conditions(albRuleCondition).actions(albRuleAction).build())
+
+       createService(ecs_cluster_name, user_name, ecsClient, albTargetGroupArn)
+
+        val userName: KeyValuePair = KeyValuePair.builder()
+                .name("user_name")
+                .value(user_name)
+                .build()
+
+        val emrClusterHostName: KeyValuePair = KeyValuePair.builder()
+                .name("emr_cluster_host_name")
+                .value(emr_cluster_host_name)
+                .build()
+
+        val chromeOverride: ContainerOverride = ContainerOverride.builder()
+                .name("headless_chrome")
+                .environment(userName)
+                .build()
+        val jupyterOverride: ContainerOverride = ContainerOverride.builder()
+                .name("jupyterHub")
+                .environment(userName, emrClusterHostName)
+                .cpu(jupyterCpu)
+                .memory(jupyterMemory)
+                .build()
+        val guacDOverride: ContainerOverride = ContainerOverride.builder()
+                .name("guacd")
+                .environment(userName)
+                .build()
+
+        val overrides: TaskOverride = TaskOverride.builder()
+                .containerOverrides(guacDOverride, chromeOverride, jupyterOverride)
+                .build()
+
+        val request: RunTaskRequest = RunTaskRequest.builder()
+                .cluster(ecs_cluster_name)
+                .launchType("EC2")
+                .overrides(overrides)
+                .taskDefinition("orchestration-service-ui-service")
+                .build()
```

Comment:
I would maybe separate out all these calls to a method and give it a meaningful name such that we know exactly what is happening here?